api/v1/storageoscluster_types.go

```diff
@@ -195,22 +195,24 @@ type ContainerImages struct {
 
 // StorageOSClusterCSI contains CSI configurations.
 type StorageOSClusterCSI struct {
-	Enable                       bool   `json:"enable,omitempty"`
-	Version                      string `json:"version,omitempty"`
-	Endpoint                     string `json:"endpoint,omitempty"`
-	EnableProvisionCreds         bool   `json:"enableProvisionCreds,omitempty"`
-	EnableControllerPublishCreds bool   `json:"enableControllerPublishCreds,omitempty"`
-	EnableNodePublishCreds       bool   `json:"enableNodePublishCreds,omitempty"`
-	EnableControllerExpandCreds  bool   `json:"enableControllerExpandCreds,omitempty"`
-	RegistrarSocketDir           string `json:"registrarSocketDir,omitempty"`
-	KubeletDir                   string `json:"kubeletDir,omitempty"`
-	PluginDir                    string `json:"pluginDir,omitempty"`
-	DeviceDir                    string `json:"deviceDir,omitempty"`
-	RegistrationDir              string `json:"registrationDir,omitempty"`
-	KubeletRegistrationPath      string `json:"kubeletRegistrationPath,omitempty"`
-	DriverRegistrationMode       string `json:"driverRegisterationMode,omitempty"`
-	DriverRequiresAttachment     string `json:"driverRequiresAttachment,omitempty"`
-	DeploymentStrategy           string `json:"deploymentStrategy,omitempty"`
+	Enable                       bool            `json:"enable,omitempty"`
+	Version                      string          `json:"version,omitempty"`
+	Endpoint                     string          `json:"endpoint,omitempty"`
+	EnableProvisionCreds         bool            `json:"enableProvisionCreds,omitempty"`
+	EnableControllerPublishCreds bool            `json:"enableControllerPublishCreds,omitempty"`
+	EnableNodePublishCreds       bool            `json:"enableNodePublishCreds,omitempty"`
+	EnableControllerExpandCreds  bool            `json:"enableControllerExpandCreds,omitempty"`
+	RegistrarSocketDir           string          `json:"registrarSocketDir,omitempty"`
+	KubeletDir                   string          `json:"kubeletDir,omitempty"`
+	PluginDir                    string          `json:"pluginDir,omitempty"`
+	DeviceDir                    string          `json:"deviceDir,omitempty"`
+	RegistrationDir              string          `json:"registrationDir,omitempty"`
+	KubeletRegistrationPath      string          `json:"kubeletRegistrationPath,omitempty"`
+	DriverRegistrationMode       string          `json:"driverRegisterationMode,omitempty"`
+	DriverRequiresAttachment     string          `json:"driverRequiresAttachment,omitempty"`
+	DeploymentStrategy           string          `json:"deploymentStrategy,omitempty"`
+	ProvisionerWorkerCount       uint8           `json:"provisionerWorkerCount,omitempty"`
+	Timeout                      metav1.Duration `json:"timeout,omitempty"`
```

Comment:
We should name this  `ProvisionerTimeout` or something, there's a lot of configurable timeouts, so it's good to be specific.

Reply:
This is timeout for all external helper, does it make sense to make configurable each of them separately?

Reply:
Yeah, it does make sense to have separate timeouts for each of them, as some containers can take much longer (the snapshother has a default timeout of 60s vs the attacher was has a default of 15s). 

It's also ok to only make the provisioner timeout configurable, as that's the one we're seeing issues with